src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -165,7 +165,11 @@ public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQ
         // FIXME: catch exceptions and implement a 500 error page here? -> in Varnish, there is a built-in error page mechanism
         if (HttpKernelInterface::MASTER_REQUEST === $type) {
             $this->traces = array();
-            $this->request = $request;
+            // Keep a duplicate of the original request for surrogates so they can access it.
+            // We must duplicate here to get a separate instance because the application will modify the request during
+            // the application flow (we know it always does because we do ourselves by setting REMOTE_ADDR to 127.0.0.1
+            // and adding the X-Forwarded-For header, see HttpCache::forward()).
+            $this->request = $request->duplicate();
```

Comment:
why duplicate and not clone? (clone is used elsewhere in the class)

Reply:
also: should we then use the duplicated/cloned request instead of $request directly?

Reply:
let's use clone then :)

Reply:
Done in dcb3833b90b8f91634223a542d0c1d91a9c04800.